### DISCUSSION TITLE
What does '_LINK_PKGS_CMD' refer to in go_generate.build_defs?

### DISCUSSION THREAD

**Discussion by PolarBear292208** [thought-machine/please#2489] [Q&A]
I'm trying to use the `go_generate` rule defined in:

https://github.com/thought-machine/pleasings/blob/master/go/go_generate.build_defs

It's failing with the following error:

```
plz-out/gen/pleasings/go/go_generate.build_defs:26:23: error: name '_LINK_PKGS_CMD' is not defined
```

I think this rule got moved from go_rules.build_defs to pleasings, then some point after that this diff removed _LINK_PKGS_CMD:

https://github.com/thought-machine/please/commit/ea9357d7338ed72e3e755426dd474fc8bc0aa848?diff=split

I think it should be replaced by `gen_import_cfg`, which is always defined as:

```
gen_import_cfg = _set_go_env() + ' && ' + _generate_pkg_import_cfg_cmd("goroot.importconfig", '"$GOROOT"')
gen_import_cfg += ' && ' + _aggregate_import_cfg_cmd()
```

What's the best way to get this into go_generate.build_defs?

**Comment by Tatskaari** [comment]
Hmm... not sure. I don't know if we can use `go generate` with the current setup reliably. As you discovered, we used to try and hack our `.a` files into locations that the top level go command would work with. The new setup uses import configs, and we've written our own variations on these commands that work more reliably for us. The top level commands like `go build` and `go generate` don't allow us to provide the import config so don't really work with the new paradigm. We should probably get rid of that rule from Pleasings. 

I always felt that the `go generate` directives didn't make that much sense when you're using a build system like Please. Can you write a `genrule()` that runs the same command from the `//go:generate` comment, or is there some magic that `go generate` provides that's non-trivial to replicate? If so, we could look into adding a `generate` subcommand to the `please_go` tool. 

Alternatively, if you're following the go workflow, you'd run `go generate` on the command line and check the result in.
